software-engineering-policies/CloudDevelopment/CloudDevelopmentPolicy.md

```diff
@@ -0,0 +1,329 @@
+| 0.1 | 07/11/2024 | Initial Draft | Cloud Guild | |
+
+### Description of Changes
+
+Version 0.1: Initial draft of Cloud Policy prepared by the Cloud Guild.
+
+## Cloud First Approach
+
+The Cloud First Approach is a strategic initiative aimed at prioritizing the adoption of cloud computing technologies in software engineering endeavors within the UK Hydrographic Office. This approach stems from the recognition of the transformative potential of cloud computing to enhance agility, innovation, and cost-effectiveness in delivering digital services.
+
+### Guiding Principles
+
+* Agility: Embrace cloud technologies to rapidly respond to changing business requirements and market dynamics.
+* Security: Ensure the confidentiality, integrity, and availability of data and applications through robust security measures.
+* Compliance: Adhere to relevant regulatory requirements and industry standards governing data privacy, security, and governance.
+* Cost Optimization: Optimize cloud resources usage to maximize cost-effectiveness and operational efficiency.
+* Innovation: Foster a culture of experimentation and innovation by leveraging cloud-native services and capabilities.
+
+### Benefits and Challenges
+
+A Cloud First Approach offers numerous benefits, including:
+
+* Enhanced scalability and flexibility
+* Faster time to market for software solutions
+* Improved collaboration and productivity
+* Cost savings through pay-as-you-go pricing models
+
+It also presents challenges such as:
+
+* Security and compliance concerns
+* Operational complexity during migration and integration
+* Potential vendor lock-in risks
+
+## Acceptable Use for Cloud Technologies
+
+The purpose of this section is to ensure that cloud resources are utilized in compliance with relevant laws, regulations, and organizational standards, while promoting efficiency, reliability, and data security.
+
+## Network Security
+
+### Guidance on Security with the Hub and Spoke Model. See [link](https://github.com/UKHO/how-do-i/blob/3373f41de95525cf795df0d9c1aae7ada2dcea86/docs/subscriptions.md?plain=1)
+
+## Recommendations for Container Hosting / Orchestration (Rich Shawley)
+
+### See Container Policy
+
+### Never Run As Root
+
+## Requirements and Recommendations for
+
+### Auditing
+
+### Testing - limit scope to infra testing ?? hand off to test community ?? - RS
+
+### Virus Scanning
+
+### Code Analysis
+
+### Dependency Checking for Cloud Applications
+
+### Infrastructure as Code - Terraform, etc
+
+Terraform is UKHO's primary tool of choise for IaC and should be used for in all cases. Terraform provided multi-cloud provisioning and so is capable of deploying to any cloud provider.
+
+#### Setting up a new project
+
+The Terraform deployment template found here: https://github.com/UKHO/Terraform-Deployment-Template should be used to ensure consistency when starting up any new project. Contact the DDC team if you have any questions.
+
+#### Networking
```

Comment:
Don't know if it is worth mentioning that Software Teams are able to send PR changes for their spokes and DDC will shephard the changes for you.